lib/libvfio-user.c

```diff
@@ -1469,6 +1657,7 @@ vfu_setup_region(vfu_ctx_t *vfu_ctx, int region_idx, size_t size,
     reg->cb = cb;
     reg->fd = fd;
     reg->offset = offset;
+    LIST_INIT(&reg->sub_reg_l);
```

Comment:
No it won't I just checked and since LIST_INIT just ensures that the value is null and the code for  REGION_IO_FDS makes a check to count how many fds it is possible to return for purposes of allocation. As the end of the list and in this case its uninitalised states are both NULL it just sets number of regions to 0 and returns no sub regions in the reply. Added a test for it and can confirm it returns a reply with no sub regions.

Reply:
You're right, it just inits to NULL anyway, thanks.

But, logically, it's the correct thing to do still. So whilst there's no actual bug here, it should still logically be moved over to where I suggested.